src/protocols/ethereum/libs/EthereumAdapter.ts

```diff
@@ -321,26 +324,81 @@ export class EthereumAdapter extends BaseProtocolAdapter {
     return normalized;
   }
 
-  override async fetchTransactions(
-    address: string,
-    nextPageParams?: string | null,
+  override async fetchAccountTransactions(
+    address: AccountAddress,
+    { nextPageNum }: ITransactionApiPaginationParams = {},
   ): Promise<IFetchTransactionResult> {
-    const { ethActiveNetworkPredefinedSettings } = useEthNetworkSettings();
-    const apiUrl = ethActiveNetworkPredefinedSettings.value.middlewareUrl;
-    const regularTransactions = await new EtherscanService(apiUrl)
-      .fetchAccountTransactions(address, nextPageParams || undefined);
+    const paginationParams: ITransactionApiPaginationParams = {};
+    let regularTransactions: ITransaction[] = [];
+
+    try {
+      const { ethActiveNetworkPredefinedSettings } = useEthNetworkSettings();
+      const service = new EtherscanService(ethActiveNetworkPredefinedSettings.value.middlewareUrl);
+      const [accountCoinTransactions, accountTokenTransactions] = await Promise.all([
+        service.fetchAccountCoinTransactions(address, { page: nextPageNum }),
+        service.fetchAccountTokenTransactions(address, { page: nextPageNum }),
+      ]);
+
+      // Remove coin fee payment transactions when the same full transaction
+      // already exists in the token transaction list.
+      accountTokenTransactions.forEach(({ hash }) => {
+        const index = accountCoinTransactions.findIndex((transaction) => transaction.hash === hash);
+        if (index > -1 && +accountCoinTransactions[index].tx.amount === 0) {
+          accountCoinTransactions.splice(index, 1);
+        }
+      });
```

Comment:
Same result can be achieved by calling `uniqBy` `hash` on regularTransactions.

Reply:
This was my initial idea but you have no control on which transaction will actually be removed. Besides the same `hash` the value (amount) is different for the coin transaction (as it pays only the fee, 0 real amount) and the token where we have both fee and amount.

Reply:
`uniqBy` will save first encounter and will remove second.